### PR TITLE
Adding `start_session` action

### PR DESCRIPTION
**Pull request by rvisuvasam**
This PR adds the `start_session` action to the driver.  It also changes the driver to use JSON only, instead of the XML default.

### CODE REVIEW

File: spec/conduit/thingspace/actions/start_session_spec.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+describe StartSession do
+  let(:request_options) do
+    {
+      oauth2_token: "89ba225e143ee95bd05c3cc2ddd1234",
+      username: "testuser",
+      password: "testpass"
+    }
+  end
+  subject { described_class.new(request_options) }
+  before { request_options[:mock_status] = "success" }
```
Comment (pedromartinez) [inline spec/conduit/thingspace/actions/start_session_spec.rb]: If you want this to be the default, you can just set it in the key/value pair in `request_options`

File: lib/conduit/thingspace/actions/base.rb [hunk 2]
```diff
@@ -14,8 +16,17 @@ def inherited(base)
     end
 
     def remote_url
-      # TODO: Define API endpoint here:
-      # http://example.com/api
+      # TODO: Should this be made configurable, and is there a better way to
+      #       construct URL.
```
Comment (ufchrisg) [inline lib/conduit/thingspace/actions/base.rb]: unless we need specific URLs for each client then I say now (drop this TODO)

Comment (pedromartinez) [reply lib/conduit/thingspace/actions/base.rb]: I agree

Comment (rvisuvasam) [reply lib/conduit/thingspace/actions/base.rb]: Fine!!!

**Comment by rvisuvasam** [comment]
Merging with confidence

**Pull request merged**